### DISCUSSION TITLE
useForm hook - Set object as value linked to Select not working

### DISCUSSION THREAD

**Discussion by danielbettech** [mantinedev/mantine#3090] [Q&A] [answered]
Hello! I'm fetching data from api and display the results in a Select component, the Select is linked to a useForm hook property in the initial values.
I am receiving an array of objects and setting the data prop like this:

```javascript
data={customers?.map((customer) => {
  return {
    value: customer,
    label: customer.name,
  }
})}
```

So the value: customer is the entire customer object, not only the id.
When I select a customer from the select, I can see the object is set in state, but the text in the Select input disappears.

If I change the value to `value: customer.id` it works as expected.

I need to retain the selected customer because I need more info from the customer to send back to the server.

**Comment by rtivital** [comment, marked as answer]
Select data value can only be string, other types are not supported (it is also not planned to support it in future).
You can retrieve item from the data array on your side:
`const selectedItem = data.find(customer => customer.id === value)`

**Reply by danielbettech** [reply]
Thanks, I will do it like this.
